src/Faker/English/FileExtension.php

```diff
@@ -0,0 +1,578 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Faker\English;
```

Comment:
Not sure about the namespace, is an implementation for German going to be different?

Reply:
I thought about this. I added it in this namespace to be constant with the German package.

The German package is called `fakerphp/german`, it lives in https://github.com/FakerPHP/German and its root namespace would be `Faker\German`. 

So there will be a `Faker\German\PersonDe` and `Faker\German\PersonAu`.. 

-------

However, The Australian English will not be in the root package. There should also be a `fakerphp/english` with Brittish, Australian... etc.

Reply:
We currently have quite a few providers that are not necessarily language specific (besides this one), such as `UserAgent` or `Barcode`.
How about adding a dedicated package/namespace that provides those?

Reply:
I have considered that and decided not to. 

It is impossible of us to know if there is a specific Russian barcode or a Chinese UserAgent. So I think the core should treat all extensions the same. But I agree with you, there are only some that are likely to be localised.

Reply:
Imo English should not be in the core package as well? Lets just put it in its own package.

Meaning people will require then locale extension and not the core directly?

Reply:
> I have considered that and decided not to. 
> 
> 
> 
> It is impossible of us to know if there is a specific Russian barcode or a Chinese UserAgent. So I think the core should treat all extensions the same. But I agree with you, there are only some that are likely to be localised. 

But arent we duplicating a lot this way? There are a lot of generic things so. Medical / Barcore / Number / Date etc

Reply:
No. We are not duplicating. 
Since the core package includes all English and “generic providers”, no language specific extensions need to duplicate anything.

Reply:
I'm somewhat reminded of our conversation in https://github.com/Nyholm/next-faker/issues/14.
These "generic providers" are in the `fakerphp/english` package or in the core? How do we differentiate between generic and "localizable"?

Reply:
Why even put english in the core package. I think it is a benefit to keep it separated and just require it. Reason for this is that it is more easy to setup new locale extensions from a proper example repo

Reply:
> Imo English should not be in the core package as well? Lets just put it in its own package.

If so, a third party library will require `fakerphp/faker` and get an unusable library because it does not have any extension. So the third partly library will then require `fakerphp/faker` **and** `fakerphp/english` and the potential "win" for separating them is lost and we only caused confusion.

> But arent we duplicating a lot this way? There are a lot of generic things so. Medical / Barcore / Number / Date etc

No, we are not. The `fakerphp/german` does not have to include a Medical extension, it will just keep using the default ones.

> These "generic providers" are in the fakerphp/english package or in the core? 

They are in the code. 

> How do we differentiate between generic and "localizable"?

Just for context (so I understand), I have a counter question: Why would you need to differentiate between generic and "localizable"?

> Reason for this is that it is more easy to setup new locale extensions from a proper example repo

Im not sure this is true. Feel free to open a new issue discussion the pros and cons. Consider that 90% of the users will only use the english locale and also see the discussions on https://github.com/Nyholm/next-faker

----------

I would also like to remind you about my PR description: 

> This PR does not mean that it is a perfect solution, but it is a step towards it.

I dont mean to turn down this discussion, but it is quite far from the purpose of this PR.